File: agent_misc/mcps/codeql-development-mcp-server/server/ql/java/tools/test/CallGraphTo/Example1.java
// COMPLIANT: Methods that should not be analyzed
class Example1 {
    void unrelated1() {
        // No calls
    }

    // NON_COMPLIANT: Target method for call graph analysis
    void targetFunc() {
        unrelated1();
    }

    void caller1() {
        targetFunc();
    }

    void caller2() {
        targetFunc();
    }
}

File: agent_misc/mcps/codeql-development-mcp-server/server/ql/csharp/tools/test/CallGraphTo/Example1.cs
// COMPLIANT: Methods that should not be analyzed
class Example1 {
    void Unrelated1() {
        // No calls
    }

    // NON_COMPLIANT: Target method for call graph analysis
    void TargetFunc() {
        Unrelated1();
    }

    void Caller1() {
        TargetFunc();
    }

    void Caller2() {
        TargetFunc();
    }
}
